tools/gsnapshot/Instances.cs

```diff
@@ -0,0 +1,312 @@
+                ComputeData.Operation response = request.Execute();
+                bool done = false;
+                if (diskType == "regions")
+                {
+                    string region = _utils.GetLastPart(disks[diskName].Region);
+                    done = await _operations.WaitForOperation(projectId, "zone", GetLastPart(instance.Zone), GetLastPart(response.SelfLink));
+
+                    _logger.LogInformation($"Adding commit lookup label to regional disk {diskName}...");
+                    var oldDisk = await _disks.GetRegionalDiskAsync(projectId, region, diskName);
+
+                    ComputeData.RegionSetLabelsRequest labelRequestBody = new ComputeData.RegionSetLabelsRequest();
+                    if (oldDisk.Labels == null)
+                    {
+                        labelRequestBody.Labels = new Dictionary<string, string>();
+                    }
+                    else
+                    {
+                        labelRequestBody.Labels = oldDisk.Labels;
+                    }
+                    labelRequestBody.Labels["gsnapshot"] = $"{snapshotPrefix}-{GetLastPart(oldDisk.Type)}-{oldDisk.Name}";
+                    labelRequestBody.LabelFingerprint = oldDisk.LabelFingerprint;
+                    RegionDisksResource.SetLabelsRequest labelRequest = computeService.RegionDisks.SetLabels(labelRequestBody, projectId, region, diskName);
+                    ComputeData.Operation labelResponse = labelRequest.Execute();
+
+                    done = await _operations.WaitForOperation(projectId, "region", region, GetLastPart(labelResponse.SelfLink));
+                }
+                else
+                {
+                    done = await _operations.WaitForOperation(projectId, "zone", zone, GetLastPart(response.SelfLink));
+
+                    _logger.LogInformation($"Adding commit lookup label to zonal disk {diskName}...");
+                    var oldDisk = await _disks.GetDiskAsync(projectId, zone, diskName);
+
+                    ComputeData.ZoneSetLabelsRequest labelRequestBody = new ComputeData.ZoneSetLabelsRequest();
+                    if (oldDisk.Labels == null)
+                    {
+                        labelRequestBody.Labels = new Dictionary<string, string>();
+                    }
+                    else
+                    {
+                        labelRequestBody.Labels = oldDisk.Labels;
+                    }
+                    labelRequestBody.LabelFingerprint = oldDisk.LabelFingerprint;
+                    labelRequestBody.Labels["gsnapshot"] = $"{snapshotPrefix}-{_utils.GetLastPart(oldDisk.Type)}-{oldDisk.Name}";
+                    DisksResource.SetLabelsRequest labelRequest = computeService.Disks.SetLabels(labelRequestBody, projectId, zone, diskName);
+                    ComputeData.Operation labelResponse = labelRequest.Execute();
+
+                    done = await _operations.WaitForOperation(projectId, "zone", zone, GetLastPart(labelResponse.SelfLink));
```

Comment:
It would be great if you could consider splitting out the request creation to separat methods each, that would increase readability a lot.